### PR TITLE
feat: Upgrade `parse_server_sdk` to `5.x.x`

### PR DESCRIPTION
**Pull request by Nidal-Bakir**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

### Issue Description

Closes: #861

### TODOs before merging

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [x] A changelog entry

### CODE REVIEW

**Comment by Nidal-Bakir** [comment]
The CI checks are falling because `parse_server_sdk: 5.0.0` has not been released yet.

**Comment by Nidal-Bakir** [comment]
@mtrezza 
The last one!

**Comment by mtrezza** [comment]
As you commented in https://github.com/parse-community/Parse-SDK-Flutter/pull/878#issuecomment-1546764559:

> But then if anyone uses Flutter SDK will be locked to Dart SDK 4.x.x.

Could we instead just say `>=4.0.0 <6.0.0` in #868?

Then #868 would not be a breaking change. I'm not sure if that makes sense given the Flutter / Dart co-dependency. But I guess we'd need to expand our CI to test the Parse Flutter SDK with the Parse Dart SDK 4.x and 5.x. Would it have any benefits for developers if we published a Parse Flutter SDK 5 that is compatible with Parse Dart SDK 4?

**Comment by Nidal-Bakir** [comment]
I'm thinking about it. 
The thing is pub will never use the 4.0.0

**Comment by Nidal-Bakir** [comment]
> Because every version of flutter_test from sdk depends on collection 1.17.0 and parse_server_sdk >=5.0.0 depends on collection ^1.17.1, flutter_test from sdk is incompatible with parse_server_sdk >=5.0.0.

And here is the answer to your question :)

**Comment by mtrezza** [comment]
Right, I think it would only make sense if a developer would need to define both dependencies, the Parse Dart SDK *and* the Parse Flutter SDK in an app. Then they could in theory keep using their Parse Dart SDK 4.x while also using the Parse Flutter SDK 5.x. But is that even a realistic scenario?

**Comment by Nidal-Bakir** [comment]
let me run some tests. that not good at all

**Comment by mtrezza** [comment]
It seems strange because `flutter_test` is just the testing framework, right? So why would the testing framework influence the Parse SDKs compatibility. Maybe we need to change the testing framework then?

**Comment by Nidal-Bakir** [comment]
If all the CIs resolve the dependency without any errors then we need to lower the collection version to : collection: ^1.16.0

**Comment by Nidal-Bakir** [comment]
It seems strange because flutter_test is just the testing framework, right? So why would the testing framework influence the Parse SDKs compatibility. Maybe we need to change the testing framework then?

We can not change the testing framework

**Comment by mtrezza** [comment]
> parse_server_sdk >=5.0.0 depends on collection ^1.17.1,

Can lower that dependency in the Parse Dart SDK and release a version 5.0.1?

**Comment by Nidal-Bakir** [comment]
All the CIs resolve the dependency without any errors when using collection: ^1.16.0. on Dart SDK I test it on my PC also using Flutter 3.3.8 and 3.10. and all good when I changed to collection: ^1.16.0

We need to release Dart 5.0.1 updating the collection dependncy to be `collection: ^1.16.0.`

**Comment by mtrezza** [comment]
Great, do you want to open a PR?

**Comment by Nidal-Bakir** [comment]
Consider it done

**Comment by Nidal-Bakir** [comment]
can you re-run the CI?

File: packages/flutter/CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
-## [5.0.0](https://github.com/parse-community/Parse-SDK-Flutter/compare/flutter-4.0.0...flutter-5.0.0) (UNRELEASED)
+## [5.0.0](https://github.com/parse-community/Parse-SDK-Flutter/compare/flutter-4.0.0...flutter-5.0.0)
```
Comment (mtrezza) [inline packages/flutter/CHANGELOG.md]:
```suggestion
## [5.0.0](https://github.com/parse-community/Parse-SDK-Flutter/compare/flutter-4.0.0...flutter-5.0.0) (2023-05-14)
```

**Comment by Nidal-Bakir** [comment]
yeah, everything is green!

File: packages/flutter/pubspec.yaml [hunk 2]
```diff
@@ -28,7 +28,6 @@ dependencies:
   sembast: ^3.4.0+6
   sembast_web: ^2.1.0+4
   path: ^1.8.2 # required for transitive use only
-  platform: ^3.1.0
```
Comment (mtrezza) [inline packages/flutter/pubspec.yaml:31]: Why has this been removed?

Comment (Nidal-Bakir) [reply packages/flutter/pubspec.yaml:31]: Not used.

Comment (mtrezza) [reply packages/flutter/pubspec.yaml:31]: Got it

**Comment by mtrezza** [comment]
Can this be merged and we can then publish the Flutter 5 release?

**Comment by Nidal-Bakir** [comment]
That's all. 

We should be good to go.

**Comment by mtrezza** [comment]
Nice, great work!

**Pull request merged**